test/Dialect/FIRRTL/canonicalization.mlir

```diff
@@ -3197,4 +3197,52 @@ firrtl.module @HasBeenReset(in %clock: !firrtl.clock, in %reset1: !firrtl.uint<1
   %constClockR1 = firrtl.node sym @constClockR1 %c5 : !firrtl.uint<1>
 }
 
+// OMIR annotations should not block removal.
+//   - See: https://github.com/llvm/circt/issues/6199
+//
+// CHECK-LABEL: firrtl.module @OMIRRemoval
+firrtl.module @OMIRRemoval(in %source : !firrtl.uint<1>) {
+  // CHECK-NOT %tmp_0
+  %tmp_0 = firrtl.wire {
+    annotations = [
+      {
+         class = "freechips.rocketchip.objectmodel.OMIRTracker",
```

Comment:
Previously the idea was that the tracker could be dropped if the signal was /optimized away/ (which is often used to mean specific notions of specific transformations)-- but if the tracker is present it could be expected it played the same role in the emitted circuit.  I'm unsure bypassing around such signals is safe or at least may need to be carefully considered.  Here it's likely the signal would be deleted later by IMDCE or other things but maybe a register we forward around would not -- or at least not /necessarily/ -- and so we may end up with tracker pointing to something that is different in a sense than it was in the input.

Maybe when this really matters different versions of tracker are used (and/or + dontTouch?), and regardless am 100% for better semantics here not defined in terms of specific optimizations and previous pipeline behavior, just.. heads-up.

Reply:
(Currently this is upheld by only dropping these if we're also deleting the module containing them)